### PR TITLE
adds related content field to DatasetDetails response dataset model

### PR DESCRIPTION
**Pull request by patrickjfl**
### What

Adds support for related content field in dataset model, for use in Cantabular metadata journey.

### How to review

Read the code and run the tests.

### Who can review

Anyone

### CODE REVIEW

File: dataset/data_test.go [hunk 1]
```diff
@@ -139,7 +146,8 @@ func expectedData(isEmpty bool) string {
 		"Related Links: [{related dataset url related dataset title}]\n" +
 		"Canonical Topic: canonicalTopicID\n" +
 		"Subtopics: [secondaryTopic1ID secondaryTopic2ID]\n" +
-		"Survey: census\n"
+		"Survey: census\n" +
+		"Related Content: [{related content description related content url related content title}]\n"
```
Comment (cookel2) [inline dataset/data_test.go:150]: I know that this is not in these changes but the writeToFile function, below this line, should probably be deleted as it is not being used.

**Review by cookel2 — COMMENTED** [review]
Just one comment but the changes look good to me.

File: dataset/data_test.go (hunk 1 above)
Comment (patrickjfl) [reply dataset/data_test.go:150]: true although from reading the comment above it it looks like it might be useful in debugging?

Comment (cookel2) [reply dataset/data_test.go:150]: Yes, I did notice the comment, but I'm of the view that code should not be left in for debugging purposes alone. Debugging comments should be removed when done with. And a function like that should be stored somewhere else, such as in documentation, if it's useful to keep.

**Review by cookel2 — APPROVED** [review]
looks good to me

**Review by vmilitaru — APPROVED** [review]

**Pull request merged**